### DISCUSSION TITLE
How do we use partition key and clustering key with Exandra?

### DISCUSSION THREAD

**Discussion by kennethtegrado** [vinniefranco/exandra#73] [Q&A] [answered]
From the documentation example we have this:
```elixir
defmodule User do
  use Ecto.Schema

  @primary_key {:id, Ecto.UUID, autogenerate: true}
  schema "users" do
    field :email, :string
    field :meta, Exandra.Map, key: :string, value: :string
  end
end
```

Based on the example, I don't see any way to use clustering key and partition key with Exandra

**Comment by vinniefranco** [comment, marked as answer]
Hello!

It's most definitely supported in migrations and schemas. Example: https://github.com/vinniefranco/exandra/blob/main/test/exandra/connection_test.exs#L820

Just looks like we're missing documentation on it.

Here's an example from our production app:

```elixir
  @primary_key false
  typed_schema "trial_redemptions" do
    field(:code_prefix, :string, primary_key: true)
    field(:code_postfix, :string, primary_key: true)
    field(:code, :string, primary_key: true)
    field(:deliver_at, :utc_datetime)
    field(:expires_at, :utc_datetime)
    field(:gift_message, :string)
    ...
```

Note: order is _important_

**Reply by kennethtegrado** [reply]
Thank you for this! To be honest, I just migrated to Phoenix that's why I am still adjusting to the many features it can offer. Thank you very much for the help!

**Reply by vinniefranco** [reply]
Hey! That's great to hear! Welcome to the club and I hope you enjoy your stay :) also, if you can - join the elixir slack. It's super friendly in there and really useful
